data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -66,20 +69,25 @@ class RemotePeerForwarder implements PeerForwarder {
                         final String pluginId,
                         final Set<String> identificationKeys,
                         final PluginMetrics pluginMetrics,
-                        final Integer batchDelay) {
+                        final Integer batchDelay,
+                        final Integer failedForwardingRequestLocalWriteTimeout,
+                        final ExecutorService forwardingRequestExecutor) {
         this.peerForwarderClient = peerForwarderClient;
         this.hashRing = hashRing;
         this.peerForwarderReceiveBuffer = peerForwarderReceiveBuffer;
         this.pipelineName = pipelineName;
         this.pluginId = pluginId;
         this.identificationKeys = identificationKeys;
         this.batchDelay = batchDelay;
+        this.failedForwardingRequestLocalWriteTimeout = failedForwardingRequestLocalWriteTimeout;
+        this.forwardingRequestExecutor = forwardingRequestExecutor;
         recordsActuallyProcessedLocallyCounter = pluginMetrics.counter(RECORDS_ACTUALLY_PROCESSED_LOCALLY);
         recordsToBeProcessedLocallyCounter = pluginMetrics.counter(RECORDS_TO_BE_PROCESSED_LOCALLY);
         recordsToBeForwardedCounter = pluginMetrics.counter(RECORDS_TO_BE_FORWARDED);
         recordsSuccessfullyForwardedCounter = pluginMetrics.counter(RECORDS_SUCCESSFULLY_FORWARDED);
         recordsFailedForwardingCounter = pluginMetrics.counter(RECORDS_FAILED_FORWARDING);
-        recordsReceivedFromPeersCounter = pluginMetrics.counter(RECORDS_RECEIVED_FROM_PEERS);
+        recordsReceivedFromPeersCount = new AtomicInteger(0);
+        pluginMetrics.gauge(RECORDS_RECEIVED_FROM_PEERS, recordsReceivedFromPeersCount);
```

Comment:
It appears that you made this a gauge so that you could decrease for all the records moved locally. Do you think there might a be a different way to avoid counting these while still using a counter?

This is technically a breaking change because are changing the metric (including the name Micrometer produces).

Reply:
Perhaps we can move the existing counter to `PeerForwarderHttpService` instead of having it here.

Reply:
I did not realize this was breaking. I'll look into moving it to the HttpService